### PR TITLE
Host Orchestrator accepts build API credentials in create CVD requests

### PR DESCRIPTION
**Pull request by jemoreira**
The credentials field is optional. If provided it will use it to directly access the build API or will pass it to `cvd fetch`.

### CODE REVIEW

**Comment by jemoreira** [comment]
Sorry for the noise, this PR isn't ready for review yet: I added a dependency that requires a newer version of Go than we're currently using.

**Comment by jemoreira** [comment]
This PR is ready for review again.

**Review by kwstephenkim — APPROVED** [review]

**Review by Databean — APPROVED** [review]

File: frontend/src/host_orchestrator/orchestrator/artifactsmanager.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	return h.syncDownload(outDir, f)
+}
+
+func (h *ArtifactsManager) GetKernelBundle(buildID, target string, fetcher ArtifactsFetcher) (string, error) {
+	f := func() (string, error) {
+		outDir := fmt.Sprintf("%s/%s_%s__kernel", h.rootDir, buildID, target)
+		if err := createDir(outDir); err != nil {
+			return "", err
+		}
+		if err := fetcher.FetchArtifacts(outDir, buildID, target, "bzImage"); err != nil {
```
[the remark this answers is not in the document]
Comment (jemoreira) [reply frontend/src/host_orchestrator/orchestrator/artifactsmanager.go:87]: acknowledged, created a bug to track fixing this.

File: frontend/src/host_orchestrator/orchestrator/artifactsmanager.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import (
+	"fmt"
+	"sync"
+)
+
+// Downloads and organizes the Build artifacts.
+//
+// Artifacts will be organized the following way:
+//  1. $ROOT_DIR/<BUILD_ID>_<TARGET>__cvd will store a full download.
+//  2. $ROOT_DIR/<BUILD_ID>_<TARGET>__kernel will store kernel artifacts only.
```
[the remark this answers is not in the document]
Comment (jemoreira) [reply frontend/src/host_orchestrator/orchestrator/artifactsmanager.go:26]: That's a very good question, but this PR is only moving the code around, not introducing it, so let's talk about this offline.

**Pull request merged**